query/src/main/java/org/infinispan/query/impl/LifecycleManager.java

```diff
@@ -134,8 +133,7 @@ private void createQueryInterceptorIfNeeded(ComponentRegistry cr, Configuration
       }
 
       ConcurrentMap<GlobalTransaction, Map<Object, Object>> txOldValues = new ConcurrentHashMap<>();
-      IndexModificationStrategy indexingStrategy = IndexModificationStrategy.configuredStrategy(searchIntegrator, cfg);
-      QueryInterceptor queryInterceptor = new QueryInterceptor(searchIntegrator, keyTransformationHandler, indexingStrategy, txOldValues, cache);
+      QueryInterceptor queryInterceptor = new QueryInterceptor(searchIntegrator, keyTransformationHandler, txOldValues, cache);
```

Comment:
It still need to read somehow the configured ```Index```, since it is deprecated, not removed. ```Index.PRIMARY_OWNER``` should probably throw an exception since it is not a safe option for non-shared indexes,and we removed all shared index managers.  When configured, ```Index.NONE``` and ```Index.ALL``` should be respected over the automatic detection

Reply:
> It still need to read somehow the configured `Index`, since it is deprecated, not removed. `Index.PRIMARY_OWNER` should probably throw an exception since it is not a safe option for non-shared indexes,and we removed all shared index managers. When configured, `Index.NONE` and `Index.ALL` should be respected over the automatic detection

yes, I've commented about that -> https://github.com/infinispan/infinispan/pull/8204#discussion_r411919976

Reply:
I still insist on respecting the config:
* Index.NONE -> don't index, log WARN
* Index.PRIMARY_OWNER -> throw error
* Index.ALL -> do nothing, log WARN

Reply:
I'd be also OK with changing the current deprecation WARN logs to mention: ``` "Index" config is deprecated and ignored, please check the upgrade guide```

Reply:
I'm perfectly fine respecting the config, but your "ISPN-11241 Deprecate Indexing mode"  basically obliterated that, and was not sure if it was a mistake or intentional :) And the later removals made it worse, turning some code into dead code. So we need to put something back there.

Reply:
It was my mistake. So why don't we settle for this PR + enriching the WARN that those config are ignored?

Reply:
I've done this, in first and second commit. And added several more commits that are ISPN-11241 floow-up